spacy/training/alignment_array.pyx

```diff
@@ -6,24 +6,26 @@ import numpy
 cdef class AlignmentArray:
     """AlignmentArray is similar to Thinc's Ragged with two simplfications:
     indexing returns numpy arrays and this type can only be used for CPU arrays.
-    However, these changes make AlginmentArray more efficient for indexing in a
+    However, these changes make AlignmentArray more efficient for indexing in a
     tight loop."""
 
     __slots__ = []
 
     def __init__(self, alignment: List[List[int]]):
-        self._lengths = None
-        self._starts_ends = numpy.zeros(len(alignment) + 1, dtype="i")
-
         cdef int data_len = 0
         cdef int outer_len
         cdef int idx
+
+        starts_ends = [0] * (len(alignment) + 1)
         for idx, outer in enumerate(alignment):
             outer_len = len(outer)
-            self._starts_ends[idx + 1] = self._starts_ends[idx] + outer_len
+            starts_ends[idx + 1] = starts_ends[idx] + outer_len
```

Comment:
Sometimes in these cases (since we are in Cython-land), it can also help to treat the numpy array as a raw C array. Since we know the length, it's safe and it avoids all the overhead of NumPy array indexing, but is probably faster than first constructing a Python list and then converting that.

Reply:
Great point! I find it's very easy to forget that C arrays are an option too :)